instance/models/mixins/secret_keys.py

```diff
@@ -0,0 +1,111 @@
+    """
+    secret_key = models.CharField(max_length=48, blank=True)
+
+    class Meta:
+        abstract = True
+
+    def set_random_key(self):
+        """
+        Generates a random seed for this instance to use when creating
+        derived secret keys for instance variables
+        """
+        self.secret_key = generate_secret_key(48)
+
+    def generate_key(self, variable_name):
+        """
+        Use the instance's secret key and the name of the variable to generate a
+        secret key specific to that variable.
+        """
+        secret_key = b64decode(self.secret_key)
+        variable_name = variable_name.encode('utf-8')
+        return hmac.new(secret_key, msg=variable_name, digestmod=hashlib.sha256).hexdigest()
+
+    def get_secret_key_settings(self):
+        """
+        Render the secret keys template as YAML and return it for use
+        in the appserver.
+        """
+        if not self.secret_key:
+            return ''
+        template = loader.get_template('instance/ansible/secret-keys.yml')
+        return template.render(
+            {
+                "ANALYTICS_API_SECRET_KEY": self.generate_key('ANALYTICS_API_SECRET_KEY'),
+                "CREDENTIALS_SECRET_KEY": self.generate_key('CREDENTIALS_SECRET_KEY'),
+                "DISCOVERY_SECRET_KEY": self.generate_key('DISCOVERY_SECRET_KEY'),
+                "ECOMMERCE_SECRET_KEY": self.generate_key('ECOMMERCE_SECRET_KEY'),
+                "ECOMMERCE_CYBERSOURCE_SECRET_KEY": self.generate_key('ECOMMERCE_CYBERSOURCE_SECRET_KEY'),
+                "ECOMMERCE_WORKER_JWT_SECRET_KEY": self.generate_key('ECOMMERCE_WORKER_JWT_SECRET_KEY'),
+                "EDX_NOTES_API_SECRET_KEY": self.generate_key('EDX_NOTES_API_SECRET_KEY'),
+                "EDXAPP_EDXAPP_SECRET_KEY": self.generate_key('EDXAPP_EDXAPP_SECRET_KEY'),
+                "EDXAPP_JWT_SECRET_KEY": self.generate_key('EDXAPP_JWT_SECRET_KEY'),
+                "EDXAPP_EDX_API_KEY": self.generate_key('EDXAPP_EDX_API_KEY'),
+                "EDXAPP_PROFILE_IMAGE_SECRET_KEY": self.generate_key('EDXAPP_PROFILE_IMAGE_SECRET_KEY'),
+                "INSIGHTS_SECRET_KEY": self.generate_key('INSIGHTS_SECRET_KEY'),
+                "NOTIFIER_LMS_SECRET_KEY": self.generate_key('NOTIFIER_LMS_SECRET_KEY'),
+                "PROGRAMS_SECRET_KEY": self.generate_key('PROGRAMS_SECRET_KEY'),
+                "ECOMMERCE_SOCIAL_AUTH_EDX_OIDC_KEY": self.generate_key('ECOMMERCE_SOCIAL_AUTH_EDX_OIDC_KEY'),
+                "ECOMMERCE_SOCIAL_AUTH_EDX_OIDC_SECRET": self.generate_key('ECOMMERCE_SOCIAL_AUTH_EDX_OIDC_SECRET'),
+                "EDXAPP_ANALYTICS_API_KEY": self.generate_key('EDXAPP_ANALYTICS_API_KEY'),
+                "FORUM_API_KEY": self.generate_key('FORUM_API_KEY'),
+                "NOTIFIER_COMMENT_SERVICE_API_KEY": self.generate_key('NOTIFIER_COMMENT_SERVICE_API_KEY'),
+                "NOTIFIER_USER_SERVICE_API_KEY": self.generate_key('NOTIFIER_USER_SERVICE_API_KEY')
```

Comment:
I don't like having these hard-coded in the app either... I was going to suggest that you could write an ansible lookup plugin, so that the final YML sent to ansible would look like
```
ANALYTICS_API_SECRET_KEY: "{{ lookup('hmac_secret', 'ANALYTICS_API_SECRET_KEY', key=secret_key_base) }}"
```

But honestly that's a lot of work, and I realized there's a much simpler solution since we're actually pre-rendering the ansible vars using Django's template engine.

If you create a class like this:

```python
class SecretKeyProvider:
    def __init__(self, secret_key):
        self.secret_key = secret_key

    def __getattr__(self, var_name):
        return hmac.new(self.secret_key, msg=var_name.encode('utf-8'), digestmod=hashlib.sha256).hexdigest()
```

Then you can write your template like this:
```
ANALYTICS_API_SECRET_KEY: '{{ secret_generator.ANALYTICS_API_SECRET_KEY }}'
CREDENTIALS_SECRET_KEY: '{{ secret_generator.CREDENTIALS_SECRET_KEY }}'
...
```

and call it like this: `template.render({"secret_generator": SecretKeyProvider(b64decode(self.secret_key))})`

I tried that to confirm it should work and produce the result you want without needing to hard-code var names.